### ISSUE TITLE
CI: operate and tasklist CI not working on PRs from forks

### ISSUE THREAD

**Issue by cmur2** [camunda/camunda#19222]
## Description

The GHA workflows from Operate & Tasklist project were developed in private repositories and never designed to be used on Pull Requests from forks by external contributors which can now happen in `camunda/camunda`. This manifests e.g. now in https://github.com/camunda/camunda/pull/18885 by instructions like https://github.com/camunda/camunda/blob/95e929f1720cb99cb0199e5cd9c41a4d6f695407/.github/workflows/tasklist-ci-build-reusable.yml#L47-L49 and also no access to secrets.

The Zeebe CI GHA workflows deal with that gracefully.

### Impact

Since Operate & Tasklist GHA workflows are part of the merge queue they represent GitHub required status checks that cannot be circumvented, so **since April 2024 no PRs from external contributors can be merged**.

#### Workaround

If a Camundi wants to merge the changes from an external contributor, the Camundi can:

1. Explain the current technical limitation to the external contributor and refer to _this_ ticket for context
2. Make sure the external contributor understands this that they will be properly credited _(Camunda does not want to steal any code, this is just for technical reasons and will be done on behalf of the contributor)_
3. Create a new PR in `camunda/camunda` explicitly crediting the external contributor, making sure to follow the [Contribution Guidelines](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md) and use the code from the original PR and backlink to it
4. Check CI status of the new PR and merge if reviewed & green

#### Possible solutions

* adjust Operate & Tasklist CI GHA workflows (minimum: those marked as GitHub required status checks) to be able to run from forks
* wait for https://github.com/camunda/camunda/issues/17721

**Comment by korthout** [comment]
For bookkeeping, this contribution is blocked by this:
- https://github.com/camunda/camunda/pull/17358#issuecomment-2157559337

**Comment by cmur2** [comment]
The 2nd contribution in June 2024 that I'm aware of and is blocked by this:

* https://github.com/camunda/camunda/pull/18885

**Comment by lenaschoenburg** [comment]
And another one: https://github.com/camunda/camunda/pull/19286

We don't want to discourage external contributions so I'd suggest to prioritize this if possible.

**Comment by saig0** [comment]
And another contribution: https://github.com/camunda/camunda/pull/16846

**Comment by korthout** [comment]
I followed the suggestion, but there are still two CI jobs failing:
- https://github.com/camunda/camunda/pull/20075

@cmur2 Is there something I did wrong, are these real failures, or does the workaround not function correctly?

**Comment by cmur2** [comment]
> I followed the suggested workaround, but there are still two (unrelated) CI jobs failing:
>     * [feat(clients/go): jobWorker can set tenantIds to activate jobs #20075](https://github.com/camunda/camunda/pull/20075)
> @cmur2 Is there something I did wrong, are these real failures, or does the workaround not function correctly?

@korthout The https://github.com/camunda/camunda/pull/20075 description says that the PR should be rebased on `main` but looking at https://github.com/camunda/camunda/tree/17167-go-client-complete-multi-tenancy-support it shows me `This branch is 5 commits ahead of, 2600 commits behind main.` - I'm not sure if I overlook something but if the PR is that outdated, it could explain not having recent fixes for CI problems?

**Comment by korthout** [comment]
@cmur2 Ah sorry about that. I thought it was rebased already 🙇

**Comment by tobiasschaefer** [comment]
My PR #24327 has been closed without merging because of this issue here....

It's kind of strange that my contribution is now blocked by this issue here (which is neither related to Tasklist nor Operate).

This will also prevent me (and others) from making more valuable contributions.

**Comment by cmur2** [comment]
> My PR #24327 has been closed without merging because of this issue here....
>
> It's kind of strange that my contribution is now blocked by this issue here (which is neither related to Tasklist nor Operate).
> 
> This will also prevent me (and others) from making more valuable contributions.

Hi, I am responsible for planning and prioritizing work around the CI in this repo and I understand your frustration re #24327 with the CI not working properly on forks.
As my coworker mentioned in https://github.com/camunda/camunda/pull/24327#issuecomment-2464376178 there is **a workaround available** to allow contributions like yours!

We are currently making progress resolving this issue e.g. via https://github.com/camunda/camunda/pull/23785 and future work on e.g. https://github.com/camunda/camunda/issues/21436 and https://github.com/camunda/camunda/issues/21437.

The reason why your contribution is blocked is that in this monorepo there is code for multiple components and these have separate CI workflows due to legacy reasons. Those were not engineered to be run on forks.

**Comment by cmur2** [comment]
https://github.com/camunda/camunda/issues/36903 broke required status checks on Pull Requests of external contributors, as they don't have access to Vault/Minimus private images for the time being.

EDIT Fixed now

**Comment by cmur2** [comment]
We don't want to intentionally break functionality of required CI checks for PRs from forks, but don't have any resources to invest beyond that (automated testing, converting optional workflows, etc) so closing this ticket for now.
